JavaScript/JavaScript.sublime-syntax

```diff
@@ -865,82 +900,62 @@ contexts:
 
   either-function-declaration:
     - match: '(?={{func_lookahead}})'
-      push: function-declaration
+      set: function-declaration
     - match: '(?={{arrow_func_lookahead}})'
-      push: arrow-function-declaration
+      set: arrow-function-declaration
 
   function-declaration:
-    - match: '\b(async)\b\s*'
-      scope: meta.function.declaration.js
-      captures:
-        1: storage.type.js
-    - match: \b(function)\b(\*)?\s*
-      scope: meta.function.declaration.js
-      captures:
-        1: storage.type.function.js
-        2: keyword.generator.asterisk.js
-    - match: '{{identifier}}'
-      scope: meta.function.declaration.js entity.name.function.js
-    - include: function-declaration-parameters
-    - match: '(?=\s*\{)'
-      set: function-block
+    - match: ''
+      set:
+        - - match: (?=\S)
+            set: function-block
+          - include: else-pop
+        - - meta_scope: meta.function.declaration.js
+          - include: immediately-pop
+        - - include: function-declaration-parameters
+          - include: else-pop
+        - - match: '{{identifier}}'
+            scope: entity.name.function.js
+            pop: true
+          - include: else-pop
+        - - match: \b(function)\b\s*(\*)?
+            captures:
+              1: storage.type.function.js
+              2: keyword.generator.asterisk.js
```

Comment:
This is also inherited from the original. If we were to change it, then either `keyword.other` or `punctuation` might be reasonable. You could even make an argument for `storage`.